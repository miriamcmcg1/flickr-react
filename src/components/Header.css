.navbar {
    background-color: #333;
    overflow: hidden;
    height: 4em;
    position: sticky;
    display: flex;
    flex-direction: row;
    vertical-align: middle;
    width: 100%;
    z-index: 1;
}

.navbar > div {
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
}

.title {
    color: #ffffff;
    font-size: 30px;
    font-weight: bold;
    margin-left: 1em;
}


.title-div > a {
    text-decoration: none;
}


@media only screen and (max-width: 750px) {
    .navbar {
        display: block;
        height: 5.5em;
    }
}
