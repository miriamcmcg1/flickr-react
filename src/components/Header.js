import React, { Component } from "react";
import SearchBar from "../containers/SearchBar";
import PropTypes from "prop-types";
import "./Header.css";
import { Link } from "react-router-dom";

class Header extends Component {
  render() {
    return (
      <div className="navbar">
        <div className="title-div">
          <Link to="/">
            <b className="title">FlickrCity</b>
          </Link>
        </div>
        {this.props.isHomePage ? <SearchBar /> : null}
      </div>
    );
  }
}

Header.propTypes = {
  isHomePage: PropTypes.bool.isRequired
};

export default Header;
