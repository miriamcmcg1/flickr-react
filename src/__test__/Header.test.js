import React from "react";
import Header from "../components/Header";
import { shallow } from "enzyme";
import "../setupTests";

describe("Header", () => {
    let props;
  beforeEach(() => {
    props = {
        isHomePage: true
    };
  });

  describe('render', () => {
      test('should render with SearchBar when isHomepage is set to true', () => {
        expect(shallow(<Header {...props}/>)).toMatchSnapshot();
      });

      test('should render without SearchBar when isHomepage is set to false', () => {
        props.isHomePage = false;
        expect(shallow(<Header {...props}/>)).toMatchSnapshot();
      });
  });
});
