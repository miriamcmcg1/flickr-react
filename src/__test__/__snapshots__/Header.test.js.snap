// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render should render with SearchBar when isHomepage is set to true 1`] = `
<div
  className="navbar"
>
  <div
    className="title-div"
  >
    <Link
      to="/"
    >
      <b
        className="title"
      >
        FlickrCity
      </b>
    </Link>
  </div>
  <ConnectFunction />
</div>
`;

exports[`Header render should render without SearchBar when isHomepage is set to false 1`] = `
<div
  className="navbar"
>
  <div
    className="title-div"
  >
    <Link
      to="/"
    >
      <b
        className="title"
      >
        FlickrCity
      </b>
    </Link>
  </div>
</div>
`;
